ui/app/serializers/pki/pki-role-engine.js

```diff
@@ -1,3 +1,77 @@
-import RoleSerializer from '../role';
+import ApplicationSerializer from '../application';
 
-export default class PkiRoleEngineSerializer extends RoleSerializer {}
+const KEY_USAGE = [
+  'digital_signature',
+  'key_agreement',
+  'key_encipherment',
+  'content_commitment',
+  'data_encripherment',
+  'cert_sign',
+  'crl_sign',
+  'encipher_only',
+  'decipher_only',
+];
+
+const EXT_KEY_USAGE = [
+  'any',
+  'server_auth',
+  'client_auth',
+  'codes_signing',
+  'email_protection',
+  'ipsec_end_system',
+  'ipsec_tunnel',
+  'time_stamping',
+  'ocsp_signing',
+  'ipsec_user',
+];
```

Comment:
These maps could potentially be moved to the model as an array of objects:
```js
keyUsage = [
  { label: 'Digital signature', value: 'digitalsignature' }
];
```
You could then loop them in the form and either add or remove the value to the `keyUsage` or `extKeyUsage` array attr on check. That would save from having to do all this work during serialization.